Core/Src/main.c

```diff
@@ -87,19 +90,25 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
-
+  HAL_TIM_PWM_Init(&htim1);
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
     /* USER CODE END WHILE */
-
+	  HAL_SPI_TransmitReceive_IT(&hspi1, &tx_buf, rx_buf, sizeof(uint8_t));
+	  *rx_buf = (uint8_t)(65535/20) + *rx_buf/20; //scales ADC digital value to be between 5-10% of 65535
+	  *rx_buf = *rx_buf & 65528; // since max value of scaled ADC value has 13 bits, we can keep most significant 10 bits
```

Comment:
Having numbers like this makes it hard to read the code, the comments are great, but in general it is good practice to use constants before the main function such as:
`static const float ADC_MAX_VALUE = 1023.0;`
That way reading the math is easier, and debugging becomes a much happier process.